types/block.go

```diff
@@ -107,6 +109,73 @@ func (b *Block) fillHeader() {
 	}
 }
 
+// // fillDataAvailabilityHeader fills in any remaining DataAvailabilityHeader fields
+// // that are a function of the block data.
+// func (b *Block) fillDataAvailabilityHeader() {
+// 	namespacedShares := b.Data.computeShares()
+// 	shares := namespacedShares.RawShares()
+// 	if len(shares) == 0 {
+// 		// no shares -> no row/colum roots -> hash(empty)
+// 		b.DataHash = b.DataAvailabilityHeader.Hash()
+// 		return
+// 	}
```

Comment:
planning to do this in a different PR, as share splitting and merging isn't added yet